### PR TITLE
Unary plus feature

### LINKED ISSUE #16073: Unary plus missing for Series and DataFrame
#### Code Sample

```
In [1]: import numpy as np

In [2]: import pandas as pd

In [3]: pd.__version__
Out[3]: '0.19.2.post+ts5'

In [4]: a = np.array([-1,0,1])  # As expected

In [5]: +a
Out[5]: array([-1,  0,  1])

In [6]: s = pd.Series(data=a)

In [7]: +s  # Unexpected
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-7-d1319861d847> in <module>()
----> 1 +s

TypeError: bad operand type for unary +: 'Series'

In [8]: d = pd.DataFrame(a)

In [9]: +d  # Unexpected
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-9-e4fd8df22471> in <module>()
----> 1 +d

TypeError: bad operand type for unary +: 'DataFrame'

In [10]: -a
Out[10]: array([ 1,  0, -1])

In [11]: -s
Out[11]: 
0    1
1    0
2   -1
dtype: int64

In [12]: -d
Out[12]: 
   0
0  1
1  0
2 -1
```
#### Problem description

Unary plus is well-defined on built-in types and Numpy ndarrays.  It is not defined for Series or DataFrame.  It's not ever necessary, but the asymmetry with the present unary minus is unexpected.  It's also occasionally nice to use it to line up a positive expression against a negative expression.  Likely adding methods implemented as an identity is sufficient.

#### Expected Output

```
In [7]: +s
Out[7]: 
0   -1
1    0
2   1
dtype: int64

In [9]: +d
Out[9]: 
   0
0  -1
1  0
2 1
```

#### Output of ``pd.show_versions()``

<details>
In [13]: pd.show_versions()

INSTALLED VERSIONS
------------------
commit: None
python: 3.5.2.final.0
python-bits: 64
OS: Linux
OS-release: 4.1.35-pv-ts2
machine: x86_64
processor: 
byteorder: little
LC_ALL: en_US.UTF-8
LANG: en_US.UTF-8
LOCALE: en_US.UTF-8

pandas: 0.19.2.post+ts5
nose: 1.3.7
pip: 9.0.1
setuptools: 27.2.0
Cython: 0.25.2
numpy: 1.11.3
scipy: 0.19.0
statsmodels: None
xarray: None
IPython: 5.1.0
sphinx: 1.5.1
patsy: None
dateutil: 2.6.0
pytz: 2016.10
blosc: None
bottleneck: None
tables: None
numexpr: None
matplotlib: 1.5.1
openpyxl: 2.4.5
xlrd: 1.0.0
xlwt: None
xlsxwriter: None
lxml: None
bs4: 4.5.3
html5lib: None
httplib2: None
apiclient: None
sqlalchemy: 1.1.9
pymysql: None
psycopg2: None
jinja2: 2.9.4
boto: None
pandas_datareader: None
</details>

### PR DESCRIPTION
**Pull request by wuisawesome**
- [ x] closes #16073  
 - [ x] tests added / passed
 - [ x] passes ``git diff upstream/master --name-only -- '*.py' | flake8 --diff``
 - [ x] whatsnew entry

### CODE REVIEW

File: pandas/core/generic.py [hunk 1]
```diff
@@ -838,8 +838,7 @@ def _set_axis_name(self, name, axis=0):
         1    2
         2    3
         >>> df.index = pd.MultiIndex.from_product([['A'], ['a', 'b', 'c']])
-        >>> df._set_axis_name(["bar", "baz"])
-                 A
+        >>> df._set_axis_name(["bar", "baz"]) A
```
Comment (shoyer) [inline pandas/core/generic.py]: This shouldn't be changed

File: pandas/tests/series/test_operators.py [hunk 2]
```diff
@@ -91,6 +91,9 @@ def check(series, other, check_reverse=False):
         check(self.ts, 5, check_reverse=True)
         check(tm.makeFloatSeries(), tm.makeFloatSeries(), check_reverse=True)
 
+    def test_pos(self):
+        assert_series_equal(+self.series, self.series)
```
Comment (shoyer) [inline pandas/tests/series/test_operators.py]: need to add a test for dataframes, too

**Review by shoyer — COMMENTED** [review]
One additional concern: unary `+` should only be defined defined for numeric types that can represent signed quantities (including timedelta). For datetime and strings, it should raise TypeError.

Finally, note that implementing this is a little more complicated that writing `+_values_from_object(self)`, because unfortunately numpy does not implement unary + correctly, either (https://github.com/numpy/numpy/pull/8967).

File: pandas/tests/series/test_operators.py (hunk 2 above)
Comment (wuisawesome) [inline pandas/tests/series/test_operators.py]: Sorry, I'm new to contributing to this Pandas. Are there any other datatypes that I need to write test cases for other than series and dataframes? 

I'll write these tests and resubmit tomorrow.

**Comment by wuisawesome** [comment]
Thanks for the feedback! I made a lot of embarrassing small mistakes in this pull request that I'll fix and resubmit.

Regarding the concern about which values the `+` operator should be defined for, why not simply define it for the widest range of values possible. It should be the responsibility of the user to ensure that their operator works correctly. This is consistent with the current implementation of `__neg__`. 

Ideally `np.array` will be fixed and this entire point will be moot.

**Comment by jorisvandenbossche** [comment]
> I'll fix and resubmit.

Just in case you don't know, for resubmitting, you just have to push to the same branch and this PR will be updated (no need to submit a new PR)

**Comment by wuisawesome** [comment]
> no need to submit a new PR

That's a bit of an issue. I was messing around and really broke that branch so I made a new one. 

I will definitely keep that in mind in the future and for now I'll link to the new branch in this thread when I submit the new PR. Hopefully it won't be too confusing if anyone looks in this thread later

**Comment by jorisvandenbossche** [comment]
You can rename your branch locally, that will work as well to update this PR

**Comment by wuisawesome** [comment]
I attempted to rename the branch but something went wrong. The new PR is here: https://github.com/pandas-dev/pandas/pull/16106

Sorry for the inconvenience.

**Pull request closed without merging**